core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -100,8 +101,8 @@ object ConsoleConsumer extends Logging {
         consumer.receive()
       } catch {
         case e: Throwable => {
-          error("Error processing message, stopping consumer: ", e)
-          consumer.stop()
+          error("Error processing message, terminating consumer process: ", e)
+          // Consumer will be stopped in shutdown hook
```

Comment:
Thanks @rajinisivaram that is a valid issue due to the semantics of ```selector.wakeup```, I think it should be fixed such that ```maybeAutoCommitOffsetsSync``` in ```KafkaConsumer.close()``` will absorb the possible exception.

I have talked with @hachikuji about the issue and we will push a hotfix patch to make it this case. Moving forward with request timeout in place ```KafkaConsumer.close()``` will still not be ever-blocking. So with that I think we can go ahead and remove ```consumer.cleanup()``` and let the new consumer implementing ```consumer.close()``` with a wakeup and then a close.